staging/src/k8s.io/cli-runtime/pkg/resource/kustomizevisitor_test.go

```diff
@@ -91,7 +91,7 @@ metadata:
     note: This is a test annotation
   labels:
     app: nginx
-  name: foo-literalConfigMap-bar-8d2dkb8k24
+  name: foo-literalConfigMap-bar-g5f6t456f5
```

Comment:
is the generated suffix expected to be stable or is it ok that this changed?

Reply:
The hashes are stable for a given major version of the software, but that changed here - the serialization scheme differs.  So it's an expected change.  is it ok?

say one uses old kustomize and new kustomize on the same cluster, which can happen because version skew is allowed.  Old kustomize creates a map with content "foo" and a deployment pointing to it.  New kustomize is used to create a new map with new content "bar", and a patch for the deployment telling it to look at the new map.

Say new kustomize hashes the new "bar" content to the same 256 bits.  So the map's name doesn't change, meaning the Deployment has no reason to load the new data, so the user doesn't get the change they want.  The chance of this happening is the chance of a `crypto/sha256` collision, and presumably can happen despite using only one version of kustomize.  

I don't see production problems here, but it will annoy people who've established test coverage that contains the names of generated maps or secrets.  We've not promised those would be stable over major releases.